README.md

```diff
@@ -394,38 +396,91 @@ The following example demonstrates overwriting a value:
 </table>
 
 ## Listening for Messages
-When creating a DataLayerHelper object, you can also specify a callback function to be called 
+When creating a `DataLayerHelper` object, you can also specify a callback function to be called 
 whenever a message is pushed onto the given dataLayer. This allows your code to be notified
 immediately whenever the dataLayer has been updated, which is a key advantage of the message
-queue approach.
+queue approach. To do so, add the function under the `listener` attribute in an options object for
+the second parameter of the `DataLayerHelper` constructor.
 
 ```js
 function listener(model, message) {
   // Message has been pushed. 
   // The helper has merged it onto the model.
   // Now use the message and the updated model to do something.
 }
-var helper = new DataLayerHelper(dataLayer, listener);
+const helper = new DataLayerHelper(dataLayer, {listener: listener});
 ```
 
-### Processing the Past
+### Listening to the Past
 Tools that are loaded onto the page asynchronously or lazily will appreciate that you can also
-opt to process message that were pushed onto the dataLayer in the past. This can be done by 
-passing true as the third parameter in the DataLayerHelper constructor.
+opt to listen to message that were pushed onto the dataLayer in the past. To do so, mark the
+`listenToPast` attribute as true in an options object for the second parameter of the `DataLayerHelper`
+constructor.
 
 ```js
 function listener(model, message) {
   // Message has been pushed. 
   // The helper has merged it onto the model.
   // Now use the message and the updated model to do something.
 }
-var helper = new DataLayerHelper(dataLayer, listener, true);
+const helper = new DataLayerHelper(dataLayer, {
+  listener: listener,
+  listenToPast: true,
+});
 ```
 
 Using this option means that your listener callback will be called once for every message that
 has ever been pushed onto the given dataLayer. And on each call to the callback, the model
 will represent the abstract model at the time of the message.
 
+### Registering Processors
+
+To register multiple processors when constructing a helper, you can use the second parameter of the
+`DataLayerHelper` constructor as an options object. Pass the processors in under the `commandProcessors`
+attribute following the data structure presented:
+
+```js
+const helper = new DataLayerHelper(dataLayer, {
+  commandProcessors: {
+    'add': [
+      function(a, b) {return {sum: a + b}},
+      function(a, b) {return {totalSum: this.get('totalSum') + a + b}},
+    ],
+    'event': function(eventName) {return {lastEvent: eventName}},
+  },
+});
```

Comment:
Arrow functions are preferred
https://google.github.io/styleguide/jsguide.html#features-functions-arrow-functions

```
const helper = new DataLayerHelper(dataLayer, {
  commandProcessors: {
    'add': [
      (a, b) => {sum: a + b},
      (a, b) => {totalSum: this.get('totalSum') + a + b},
    ],
    'event': (eventName)  => {lastEvent: eventName},
  },
});
```

Reply:
@Alex7Li For registered processors to use this they have to be non-arrow functions, correct? I could switch to arrows for the ones not accessing the model via this if that is what you mean?

Reply:
You are right, the way it is is fine

Reply:
Also, be careful when the return type is an object literal.  The {} get interpreted as a block scope, rather than an object definition, and you have to wrap them in parens to get the expected behavior.

```js
const foo = (a) => {sum: a + 2};
console.log(foo(1));
const bar = (a) => ({sum: a + 2});
console.log(bar(1));
// Output:
// undefined
// {sum: 3}